operator/pkg/bgpv2/manager.go

```diff
@@ -91,19 +99,42 @@ func registerBGPResourceManager(p BGPParams) *BGPResourceManager {
 		health:    p.Health,
 		metrics:   p.Metrics,
 
-		reconcileCh:        make(chan struct{}, 1),
-		bgpClusterSyncCh:   make(chan struct{}, 1),
-		clusterConfig:      p.ClusterConfigResource,
-		nodeConfigOverride: p.NodeConfigOverrideResource,
-		nodeConfig:         p.NodeConfigResource,
-		peerConfig:         p.PeerConfigResource,
-		ciliumNode:         p.NodeResource,
-
+		reconcileCh:           make(chan struct{}, 1),
+		bgpClusterSyncCh:      make(chan struct{}, 1),
+		bgpRouterIDMap:        make(map[string]*netip.Addr),
+		clusterConfig:         p.ClusterConfigResource,
+		nodeConfigOverride:    p.NodeConfigOverrideResource,
+		nodeConfig:            p.NodeConfigResource,
+		peerConfig:            p.PeerConfigResource,
+		ciliumNode:            p.NodeResource,
 		enableStatusReporting: p.DaemonConfig.EnableBGPControlPlaneStatusReport,
 	}
+	if p.DaemonConfig.BGPRouterIDAllocationMode == option.BGPRouterIDAllocationModeIPPool {
+		ipnet, err := netip.ParsePrefix(p.DaemonConfig.BGPRouterIDAllocationIPPool)
+		if err != nil {
+			b.logger.Error(err.Error())
+			b.health.Degraded("BGP manager health degraded", err)
```

Comment:
BGP being degraded is somewhat implied from the auto generated health reporter ID plus the fact that the reporter is set to degraded, perhaps we should provide some context on the degradation? (ex. "bad BGP router allocation pool").

Reply:
I have added more info on the err itself. So it's err log and "BGP manager health degraded" with the err now